spec/code-snippets/DeclarationReferences.ts

```diff
@@ -0,0 +1,443 @@
+//    * NOT CORRECT: {@link methodA}
+//    */
+//   public methodB(): void {
+//   }
+// }
+//
+// Requiring fully scoped names ensures that documentation processors can resolve links
+// efficiently and without access to a compiler analysis.  (For this document, it means
+// all our examples can be simple self-references, since cross-references would look the same.)
+//
+// The most interesting feature of this syntax is the "[]" selector, which has these properties:
+//
+// - It is used to disambiguate an already traversed node (e.g. choosing a function overload)
+//
+// - It is NEVER used to traverse into a new node (e.g. we write MyClass."member with spaces"
+//   instead of MyClass["member with spaces"]
+//
+// - For classes, the supported selectors are "[instance]", "[static]", or "[constructor]"
+//
+// - For merged declarations, the supported selectors are "[class]", "[enum]", "[function]",
+//   "[interface]", "[namespace]", "[type]", or "[variable]"
+//
+// - When selecting a user-defined label (e.g. "[WITH_NUMBERS]") the label must be
+//   all capitals, which ensures that it cannot conflict with a TSDoc reserved name such
+//   as "class".
+
+
+//---------------------------------------------------------
+// Static vs instance members
+
+/**
+ * Shortest name:  {@link Class00}
+ * Full name:      {@link Class00[class]}
+ */
+export class Class00 {
+  /**
+   * Shortest name:  {@link Class00.member01[instance]}
+   * Full name:      {@link Class00[class].member01[instance]}
+   */
+  public member01(): void {
+  }
+
+  /**
+   * Shortest name:  {@link Class00.member01[static]}
+   * Full name:      {@link Class00[class].member01[static]}
+   */
+  public static member01(): void {
+  }
+
+  /**
+   * Shortest name:  {@link Class00[constructor]}
+   * Full name:      {@link Class00[constructor]}
+   *
+   * NOTE: Class00.constructor is NOT correct, because the constructor is NOT
+   * a member of Class00.
+   */
+  public constructor() {
+    console.log('Constructed Class00');
+  }
+
+  /**
+   * Shortest name:  {@link Class00[class].constructor}
+   * Also valid:     {@link Class15."constructor"}
+   * Full name:      {@link Class00[class].constructor[instance]}
+   *
+   * NOTE: This is NOT the class constructor, but rather a property
+   * whose name confusingly uses a keyword.
+   */
+  public get constructor(): string {
+    return 'hello';
+  }
+}
+
+//---------------------------------------------------------
+// Nesting namespaces
+
+/**
+ * Shortest name:  {@link N02.N03.N04}
+ * Full name:      {@link N02[namespace].N03[namespace].N04[namespace]}
+ */
+export namespace N02.N03.N04 {
+  /**
+   * Shortest name:  {@link N02.N03.N04.function05}
+   * Full name:      {@link N02[namespace].N03[namespace].N04[namespace].function05[function]}
+   */
+  export function function05(): void {
+  }
+}
+
+//---------------------------------------------------------
+// Function overloads
+
+/**
+ * Shortest name:  {@link function06[1]}
```

Comment:
>1 [](start = 37, length = 1)

It would be helpful to warn against using this if you're talking about your own package (perhaps in strict mode)